### DISCUSSION TITLE
Connecting a Route 53 domain

### DISCUSSION THREAD

**Discussion by valk** [vercel/vercel#4243] [General] [answered]
Hi, I'd like to connect my domain, and I changed the NS records for the domain from the defaults in Route 53 settings.
I didn't add an A record because there is no option for that.

I added another subdomain's A record that points to a **different** IP and added another NS record for it with the default Route 53 name servers.

EDIT: in this case I having a trouble to understand whether I have to change the SOA record or not. So for now, I leave it as is. But the error I'm getting from IntoDNS is [as described here](https://www.cloudns.net/wiki/article/203/).

I know that there is 48 hour domain propagation time, but I'm getting errors that seem to be irrelevant to this case.

Any help will be greatly appreciated!

Here's the IntoDNS [analysis](https://intodns.com/entheog.com).

**Comment by coetry** [comment, marked as answer]
Hi @valk there are a few ways to connect your Route 53 domains to your Vercel project.

- Change to Vercel nameservers 
- Add a CNAME record for `www` with a value of `alias.zeit.co` (will be `cname.vercel-dns.com` on [May 20th](https://www.notion.so/vercel/Change-Timeline-fab7622780694753a92c0a396cea7eb6)) and a redirect from your apex domain to the `www` subdomain

We'll be adding support for adding an A record with a static IP, but we don't currently support that. 

```sh
▲ ~ dig ns entheog.com +short
f.zeit-world.net.
c.zeit-world.org.
e.zeit-world.com.
b.zeit-world.co.uk.

```
It looks like your domain is using our nameservers correctly now. DNS propagation will always be a part of the equation and there's not really much anyone can do to avoid that when changing from other providers. 

Please reach out to us at support@vercel.com so we can review the specifics of your case if you have DNS issues in the future.

**Reply by valk** [reply]
Hi @coetry , thanks for the answer, indeed now everything works properly. 
I figured out what was the problem. From the docs it wasn't initially clear to me whether I should update the Name Servers in AWS or to add NS records (which I made initially). So when updating the former, everything started to work as expected. 
Plus, I successfully configured the A record like so:
`now dns add mydomain.com xyz A 123.123.123.123 `
so xyz.mydomain.com is now pointing to 123.123.123.123. 
Hope this reply helps others with similar issues.

**Reply by patjakubik** [reply]
I will also add that if you want to redirect your Route 53 apex domain to www subdomain you will need to:

- create S3 bucket with website hosting
- create redirect
- issue SSL cert
- create cloudfront distribution
- set your A record to cloudfront URL (as an alias)

And that will make your website redirect from http://example.com and https://example.com to https://www.example.com

More about that here: https://medium.com/faun/redirect-naked-domain-to-www-with-https-on-aws-route-53-e620e4ce989d

Also, don't forget to change all "https://" to "https://www." in your project.

It's a pain but looks like the only way to do this.

Fingers crossed for A records in the future!

**Comment by codeyourwayup** [comment]
Problem solved. 
You can use the latest Vercel feature, which is to use an A record that points to the dedicated IP of 76.76.21.21, however, in route 53, remember to use "blank" instead of "@". because in aws, it refers to "blank".mydomain.com as the apex, rather than the instruction on vercel, where it shows "@.mydomain.com" as an A record.

**Reply by JoonsungUm** [reply]
> remember to use "blank" instead of "@".

thank you!

**Reply by codeyourwayup** [reply]
Yes. It took me several hours to figure it out, even with the @, vercel still works but not in a stable way...that is why I took me such long time to get it. If it just didn't work, I might had quickly figured it out.

**Reply by mrunal-modi** [reply]
Just tried what vercel recommends i.e. created below entries in AWS Route53 but doesn't work for me - 
A record => 76.76.21.21
CNAME www => cname.vercel-dns.com.

Did not change the Name servers in AWS to below as it is optional am guessing? 
ns1.vercel-dns.com
ns2.vercel-dns.com

**Reply by jhaharsh03** [reply]
Hi, I added one CName entry and A name for vercel in my record in route53 hosted domain and I also have two other default records by AWS as it is, still, my site is not loading.

**Reply by jhaharsh03** [reply]
Do I need to delete default AWS records?
